File: codewars/6kyu/Are they the "same"?.js
function comp(a, b) {
  if (a == null || b == null) return false;

  if (a.length !== b.length) return false;

  const aSquared = a.map(x => x * x);

  aSquared.sort((x, y) => x - y);
  b.sort((x, y) => x - y);

  for (let i = 0; i < a.length; i++) {
    if (aSquared[i] !== b[i]) return false;
  }

  return true;
}

console.log(comp([121, 144, 19, 161, 19, 144, 19, 11], [121, 14641, 20736, 361, 25921, 361, 20736, 361]));

File: codewars/6kyu/Are they the "same"?.py
import numpy as np


def comp(array1, array2):
    if array2 is None:
        return False
    if array1 is None:
        return False
    if len(array1) != len(array2):
        return False
    array1 = [np.abs(a) for a in array1]
    array2 = [np.abs(a) for a in array2]
    return all(a1 ** 2 == a2 for a1, a2 in zip(sorted(array1), sorted(array2)))
